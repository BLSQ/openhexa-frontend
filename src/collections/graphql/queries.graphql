query CollectionsPage($page: Int = 1, $perPage: Int = 15) {
  collections(page: $page, perPage: $perPage) {
    ...CollectionsTable_page
  }

  me {
    authorizedActions
  }
}

query CollectionPage($id: String!) {
  collection(id: $id) {
    id
    name
    createdAt
    updatedAt
    description
    author {
      displayName
    }
    countries {
      code
      ...CountryBadge_country
    }

    tags {
      id
      ...Tag_tag
    }

    elements {
      items {
        ...CollectionElementsTable_element
      }
    }

    ...CollectionActionsMenu_collection
  }
}
