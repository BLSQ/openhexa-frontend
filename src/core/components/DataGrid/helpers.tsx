import * as React from "react";
import { Cell } from "react-table";

export const CellContext = React.createContext<Cell<any> | null>(null);

export function useCellContext() {
  const cell = React.useContext(CellContext);
  if (!cell) {
    throw new Error("useCellContext muse be used inside a CellContext wrapper");
  }
  return cell;
}

export function CellContextProvider(props: {
  cell: Cell;
  children: React.ReactNode;
}) {
  return (
    <CellContext.Provider value={props.cell}>
      {props.children}
    </CellContext.Provider>
  );
}
