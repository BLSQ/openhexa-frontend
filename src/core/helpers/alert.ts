export function displayAlert(
  message: string,
  type: "error" | "info" | "warning" = "info"
) {
  if (!window) {
    console.warn("displayAlert cannot be called on the server.");
    return;
  }
  const event = new CustomEvent("displayAlert", {
    detail: {
      message,
      type,
    },
  });

  window.dispatchEvent(event);
}
