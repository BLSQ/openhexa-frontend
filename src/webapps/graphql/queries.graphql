query WebappsPage($workspaceSlug: String!, $page: Int, $perPage: Int = 15) {
  # TODO : rename with Workspace
  workspace(slug: $workspaceSlug) {
    slug
    name
    ...WorkspaceLayout_workspace
  }

  webapps(workspaceSlug: $workspaceSlug, page: $page, perPage: $perPage) {
    totalPages
    totalItems
    items {
      id
      name
      description
      url
      isFavorite
      permissions {
        update
        delete
      }
    }
  }
}
# TODO
#query WebappPage($id: UUID!) {
#  webapp(id: $id) {
#    id
#    name
#    description
#    url
#    isFavorite
#    createdBy {
#      id
#      displayName
#    }
#    workspace {
#      id
#      name
#    }
#    permissions {
#      update
#      delete
#    }
#  }
#}
