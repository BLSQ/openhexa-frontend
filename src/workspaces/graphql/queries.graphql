query WorkspacesPage($page: Int, $perPage: Int) {
  workspaces(page: $page, perPage: $perPage) {
    totalItems
    items {
      id
      name
      countries {
        code
        flag
      }
    }
  }
}

query WorkspacePage($id: String!) {
  workspace(id: $id) {
    id
    name
    description
    countries {
      code
      flag
    }
    memberships {
      totalItems
      items {
        id
        role
        user {
          email
        }
        createdAt
      }
    }
  }
}
