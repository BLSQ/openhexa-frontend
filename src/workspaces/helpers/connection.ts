import { WorkspaceConnectionType } from "graphql-types";

export const TYPES = [
  {
    value: WorkspaceConnectionType.Postgresql,
    label: "PostgreSQL",
    color: "bg-blue-300",
    iconSrc: "/static/connector_postgresql/img/symbol.svg",
  },
  {
    value: WorkspaceConnectionType.S3,
    label: "Amazon S3 Bucket",
    color: "bg-orange-200",
    iconSrc: "/static/connector_s3/img/symbol.svg",
  },
  {
    value: WorkspaceConnectionType.Gcs,
    label: "Google Cloud Bucket",
    color: "bg-blue-200",
    iconSrc: "/static/connector_gcs/img/symbol.svg",
  },
  {
    value: WorkspaceConnectionType.Custom,
    label: "Custom",
    color: "bg-gray-200",
  },
  // {
  //   value: "dhis2",
  //   label: "DHIS2 Instance",
  //   iconSrc: "/static/connector_dhis2/img/symbol.svg",
  // },
];
