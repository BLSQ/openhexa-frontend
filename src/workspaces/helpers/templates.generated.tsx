import * as Types from '../../graphql/types';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type UpdateWorkspaceTemplateMutationVariables = Types.Exact<{
  input: Types.UpdateTemplateInput;
}>;


export type UpdateWorkspaceTemplateMutation = { __typename?: 'Mutation', updatePipelineTemplate: { __typename?: 'UpdateTemplateResult', success: boolean, errors: Array<Types.UpdateTemplateError>, template?: { __typename?: 'PipelineTemplate', id: string, name: string, description?: string | null, config?: string | null } | null } };


export const UpdateWorkspaceTemplateDocument = gql`
    mutation UpdateWorkspaceTemplate($input: UpdateTemplateInput!) {
  updatePipelineTemplate(input: $input) {
    success
    errors
    template {
      id
      name
      description
      config
    }
  }
}
    `;
export type UpdateWorkspaceTemplateMutationFn = Apollo.MutationFunction<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>;

/**
 * __useUpdateWorkspaceTemplateMutation__
 *
 * To run a mutation, you first call `useUpdateWorkspaceTemplateMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useUpdateWorkspaceTemplateMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [updateWorkspaceTemplateMutation, { data, loading, error }] = useUpdateWorkspaceTemplateMutation({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useUpdateWorkspaceTemplateMutation(baseOptions?: Apollo.MutationHookOptions<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>(UpdateWorkspaceTemplateDocument, options);
      }
export type UpdateWorkspaceTemplateMutationHookResult = ReturnType<typeof useUpdateWorkspaceTemplateMutation>;
export type UpdateWorkspaceTemplateMutationResult = Apollo.MutationResult<UpdateWorkspaceTemplateMutation>;
export type UpdateWorkspaceTemplateMutationOptions = Apollo.BaseMutationOptions<UpdateWorkspaceTemplateMutation, UpdateWorkspaceTemplateMutationVariables>;