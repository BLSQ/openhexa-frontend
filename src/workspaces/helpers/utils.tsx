import Button from "core/components/Button";
import Select from "core/components/forms/Select";
import { PipelineNotificationEvent } from "graphql/types";
import { i18n } from "next-i18next";
import { ReactElement, ReactNode } from "react";

export const formatNotificationEvent = (status: PipelineNotificationEvent) => {
  switch (status) {
    case PipelineNotificationEvent.AllEvents:
      return i18n!.t("All events");
    case PipelineNotificationEvent.PipelineFailed:
      return i18n!.t("Pipeline failed");
  }
};

interface IActionButton {
  icon?: ReactElement;
  onClick?: () => void;
  disabled?: boolean;
  render?: () => ReactNode;
}
interface ActionButtonGroupProps {
  buttons: IActionButton[];
}

export const ActionButtonGroup = ({ buttons }: ActionButtonGroupProps) => {
  return (
    <div className="flex gap-2">
      {buttons.map((button: IActionButton, index) =>
        button.render ? (
          <div key={index}>{button.render()}</div>
        ) : (
          <Button
            key={index}
            onClick={button.onClick}
            size="sm"
            variant="secondary"
            disabled={button.disabled}
          >
            {button.icon}
          </Button>
        ),
      )}
    </div>
  );
};

export const NotificationEventSelect = ({
  value,
  onChange,
}: {
  value?: PipelineNotificationEvent | null | undefined;
  onChange: (notificationEvent: PipelineNotificationEvent) => void;
}) => {
  return (
    <Select
      value={value || null}
      displayValue={(value) => formatNotificationEvent(value)}
      placeholder={i18n!.t("Select event")}
      onChange={onChange}
      getOptionLabel={(option) => formatNotificationEvent(option)}
      options={[
        PipelineNotificationEvent.AllEvents,
        PipelineNotificationEvent.PipelineFailed,
      ]}
    />
  );
};
