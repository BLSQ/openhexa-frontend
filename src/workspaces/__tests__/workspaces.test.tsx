import { render, screen } from "@testing-library/react";
import { TestApp } from "core/helpers/testutils";
import WorkspacePage from "pages/workspaces/[workspaceId]";
import { WorkspacePageDocument } from "workspaces/graphql/queries.generated";

describe("Workspaces", () => {
  it("renders the workspaces page", async () => {
    const id = "a303ff37-644b-4080-83d9-a42bd2712f63";
    const graphqlMocks = [
      {
        request: {
          query: WorkspacePageDocument,
          variables: {
            id,
          },
        },
        result: {
          data: {
            workspace: {
              id: id,
              name: "Rwanda Workspace",
              description: "This is a description",
              countries: [],
              memberships: {
                totalItems: 0,
                items: [],
              },
            },
          },
        },
      },
    ];

    const { container } = render(
      <TestApp mocks={graphqlMocks}>
        <WorkspacePage page={1} perPage={1} workspaceId={id} />
      </TestApp>
    );
    const elm = await screen.findByText("Rwanda Workspace", { selector: "a" });
    expect(container).toMatchSnapshot();
  });
});
