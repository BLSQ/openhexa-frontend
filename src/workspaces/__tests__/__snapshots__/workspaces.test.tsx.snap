// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspaces renders the workspaces page 1`] = `
<div>
  <div
    class="sticky top-0 z-10 flex h-16 flex-shrink-0 items-center justify-between border-b border-gray-200 bg-white py-3 shadow px-4 md:px-6 xl:px-10 2xl:px-12"
  >
    <div
      class="flex-1 flex items-center justify-between"
    >
      <nav
        aria-label="Breadcrumbs"
      >
        <ol
          class="flex items-center"
          role="list"
        >
          <li
            class=""
          >
            <div
              class="flex items-center"
            >
              <span
                class="text-sm hover:text-gray-700 font-medium text-gray-500"
              >
                <a
                  class=""
                  href="/workspaces/a303ff37-644b-4080-83d9-a42bd2712f63"
                >
                  Rwanda Workspace
                </a>
              </span>
            </div>
          </li>
        </ol>
      </nav>
      <button
        class="relative shadow-sm rounded h-fit
    focus:outline-none focus:ring-2 focus:ring-offset-2
    inline-flex items-center justify-center transition-all
    border font-medium
    disabled:opacity-50 disabled:cursor-not-allowed
   text-sm px-3 py-2.5 leading-4 border-transparent text-white bg-blue-600 hover:bg-blue-700 focus:ring-blue-500"
      >
        Edit
      </button>
    </div>
  </div>
  <div
    class="py-6 xl:py-8"
  >
    <div
      class="mx-auto px-4 md:px-6 xl:px-10  2xl:px-12"
    >
      <article
        class="sm:rounded-lg overflow-hidden shadow border-b border-gray-200 bg-white"
      >
        <section
          class="px-4 py-5 sm:px-6"
        >
          <div>
            <div
              class="prose prose-headings:font-medium max-w-3xl"
            >
              <p>
                This is a description
              </p>
            </div>
          </div>
        </section>
      </article>
    </div>
  </div>
</div>
`;
